### PR TITLE
[1494, 1498] Adding unit tests for backend component and replacing GitHub Autobuild

### PR DESCRIPTION
**Pull request by phongcao**
- Replacing GitHub autobuild with manual build steps so that we have more control over the building/testing process and speed up build time (from 4 minutes to 40 seconds). Note that we'll be adding back code scan/analysis later in one of our coming user story.
- Adding unit tests for backend components.
- Adding test visualization for backend components.

In collaboration with @nexilus18

### CODE REVIEW

File: RetrospectiveExtension.Backend.Tests/ReflectHubTests.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+using System.Threading;
+using Xunit;
+using static ReflectBackend.ReflectBackendSignals;
+
+namespace RetrospectiveExtension.Backend.Tests
+{
+  public class ReflectHubTests
+  {
+    [Fact]
+    public async void BroadcastDeletedBoardTest()
```
Comment (NarmathaBala) [inline RetrospectiveExtension.Backend.Tests/ReflectHubTests.cs:13]: @phongcao Wanted to bring your attention to the PR @mhassaninmsft put in for updates to same code

**Review by NarmathaBala — APPROVED** [review]
Phong, looks good, please see my comments about another PR before merging

File: RetrospectiveExtension.Backend.Tests/ReflectHubTests.cs (hunk 1 above)
Comment (phongcao) [reply RetrospectiveExtension.Backend.Tests/ReflectHubTests.cs:13]: @NarmathaBala Thanks for letting me know. I guess either of us needs to resolve conflicts before merging.

**Review by polatengin — APPROVED** [review]
perfect, thanks!.. 🚀💯

**Pull request merged**